cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -119,7 +119,7 @@ protected void configure() {
 
   @Override
   public Module getDistributedModules() {
-    return getDistributedModules(false);
+    return getDistributedModules(true);
```

Comment:
I think it was done this way so that:
1. In distributed mode, always bind to ``KafkaMetadataPublisherTest``, irrespective of whether publishing is enabled using cConf.
2. In tests, bind to ``KafkaMetadataPublisherTest`` if the test needs in-memory kafka, else bind to ``NoOpMetadataPublisher``.

However, 1. is not necessary, we can also bind to ``NoOpMetadataPublisher`` in distributed mode, based on cConf like @chtyim suggested. That way we can perhaps remove the if conditions  for checking cConf in ``DefaultMetadataChangePublisher`` ?